devicetypes/smartthings/vision-in-wall-2relays-switch.src/vision-in-wall-2relays-switch.groovy

```diff
@@ -0,0 +1,279 @@
+		parameterMap.each {
+			input (title: it.name, description: it.description, type: "paragraph", element: "paragraph")
+			switch(it.type) {
+				case "boolean":
+					input(type: "paragraph", element: "paragraph", description: "Option enabled: ${it.activeDescription}\n" +
+							"Option disabled: ${it.inactiveDescription}"
+					)
+					input(name: it.key, type: "boolean", title: "Enable", defaultValue: it.defaultValue == it.activeOption, required: false)
+					break
+					
+				case "enum":
+					input(name: it.key, title: "Select", type: "enum", options: it.values, defaultValue: it.defaultValue, required: false)
+					break
+			}	
+		}
+	}
+}
+
+def installed() {
+	def componentLabel
+	if (device.displayName.endsWith('1')) {
+		componentLabel = "${device.displayName[0..-2]}2"
+	} else {		
+		componentLabel = "$device.displayName 2"
+	}
+	try {
+		String dni = "${device.deviceNetworkId}-ep2"
+		addChildDevice("Z-Wave Binary Switch Endpoint", dni, device.hub.id,
+			[completedSetup: true, label: "${componentLabel}", isComponent: false])
+		log.debug "Endpoint 2 (Z-Wave Binary Switch Endpoint) added as $componentLabel"
+	} catch (e) {
+		log.warn "Failed to add endpoint 2 ($desc) as Z-Wave Binary Switch Endpoint - $e"
+	}
+	// Preferences template
+	state.currentPreferencesState = [:]
+	parameterMap.each {
+		state.currentPreferencesState."$it.key" = [:]
+		state.currentPreferencesState."$it.key".value = getPreferenceValue(it)
+		def preferenceName = it.key + "Boolean"
+		settings."$preferenceName" = true
+		state.currentPreferencesState."$it.key".status = "synced"
+	}
+	firstCommand()
+}
+
+def firstCommand(){
+	def endpointNumber = 1
+	delayBetween([
+		encap(endpointNumber, zwave.configurationV1.configurationGet(parameterNumber: 0x01).format()),
+		encap(endpointNumber, zwave.multiChannelAssociationV2.multiChannelAssociationSet(groupingIdentifier: 0x01, nodeId: [0x01, 0x01]).format())
+	])	
+}
+
+def updated() {
+	parameterMap.each {	
+		if (isPreferenceChanged(it)) {
+			log.debug "Preference ${it.key} has been updated from value: ${state.currentPreferencesState."$it.key".value} to ${settings."$it.key"}"
+			state.currentPreferencesState."$it.key".status = "syncPending"
+		} else if (!state.currentPreferencesState."$it.key".value) {
+			log.warn "Preference ${it.key} no. ${it.parameterNumber} has no value. Please check preference declaration for errors."
+		}
+	}
+	configure()
+}
+
+def configure() {
+	// Device-Watch simply pings if no device events received for checkInterval duration of 32min = 2 * 15min + 2min lag time
+	sendEvent(name: "checkInterval", value: 30 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	def commands = []
+	parameterMap.each {
+		if (state.currentPreferencesState."$it.key".status == "syncPending") {
+			commands << zwave.configurationV1.configurationSet(scaledConfigurationValue: getCommandValue(it), parameterNumber: it.parameterNumber, size: it.size).format()
```

Comment:
Since encap() method is defined. Do You want to use it here and in a whole file ?

Reply:
This old product have not all command support multiChannelCmdEncap,
configuration and multiChannelAssociation  command class are need hold base type,
so I review and fix it in a whole file, just support multiChannelCmdEncap command call encap().